xrpl/constants.py

```diff
@@ -42,6 +42,8 @@ class XRPLException(Exception):
 :meta private:
 """
 
+HEX_MPTID_REGEX: Final[Pattern[str]] = re.compile(r"^[0-9A-Fa-f]{48}$")
```

Comment:
I was under the belief that `mpt_issuance_id` is represented by 192-bits, as explained here: https://github.com/XRPLF/rippled/blob/6cf37c4abef80fb535d1d7577a548665e139f635/include/xrpl/protocol/MPTIssue.h#L28

However, the size of `mpt_issuance_id` is much larger than 192 bits in the examples. It's four times larger in the RPC outputs.

1. Where can I find the cpp implementation of this `MPTCurrency` type?
2. Which transactions use the `MPTCurrency` type? I'd like to test the correctness of serialization/de-serialization of the `MPTCurrency` type.

Reply:
192 bits is 24 bytes, which is a hex string of length 48.

1. `MPTCurrency` is equivalent to `MPTIssue` in rippled.
2. Not sure there are any. The only two features that use `STIssue` types are XChainBridge and AMM and I don't think either of them support MPTs right now (though support for AMM is being added in [XLS-82d](https://github.com/XRPLF/XRPL-Standards/discussions/231) in https://github.com/XRPLF/rippled/pull/5285).

Reply:
1. Can we retain the same name `MPTIssue` in the client libraries. Is there a reason you prefer `MPTCurrency` ?
2. I wish there were a way to test the correctness of this serialization. Unless we communicate a serialized `MPTIssue` value to rippled, we can not validate the correctness of the client library binary codec.

As a new feature, if rippled could expose an `echo(serialized_input) => deserialized_output` -like of utility command, that would have been helpful. I'm sure there are other `SField`s which haven't been used in transactions yet.

> 192 bits is 24 bytes, which is a hex string of length 48.

Hmm yes, you are right. I had mis-understood the calculation.

Reply:
1.`MPTCurrency` aligns more with what we have in the client library and makes more sense in English. I don't have a strong opinion on this, though.
2. rippled can serialize transactions via the `sign` and `submit` fields locally. You could try with the XLS-82d branch above.

Reply:
Alright, I would slightly prefer the name `MPTIssue` over `MPTCurrency`, it aligns well with the documentation too.

The MPT amendment has been merged into the rippled codebase, I should be able to use the develop branch to test the `sign`, `submit` commands right?

> rippled can serialize transactions via the sign and submit fields locally

Hmm, this will help me with serialization, but not the other way around i.e. I'm trying to understand how I can deserialize a blob of `MPTIssue` and verify that the xrpl-py library does it correctly?